Automattic/jetpack
# Update/blocks contact form discoverability

Fixes #10903

## Proposed changes:

This change will alow registering Contact Form Block even if the Contact Form module is disabled. It should show the placeholder allowing to enable the module in one click. This is aimed to make our blocks more discoverable.

### Other information:
There's an issue when the module is not active and we are showing placeholder, we would probably want to have AI Assistant disabled. In current status, `useModuleStatus` logic won't help us sync different parts of code. We would either want to have some control over the Assistant from within block it is working on (eg from Contact Form block we could say to to show AI assistant), or, create a proper `store`, where the status will be handled. @jeherve, @retrofox what do you think?


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Spin up the branch locally and tunnel with JT to your sandbox
* Build `packages/forms` and `plugins/jetpack`
* Navigate to `/wp-admin/admin.php?page=jetpack_modules` and make sure that Contact Form module is disabled
* Create a new post `/wp-admin/post-new.php`
* In blocks menu make sure you see the `Form` block and are able to add it to the page. It will have loading state first

<img width="680" alt="Screenshot at Sep 14 15-35-30" src="https://github.com/Automattic/jetpack/assets/60262784/c7ed4c14-19b6-43d3-8d5b-f35960d4232c">

<img width="663" alt="Screenshot at Sep 14 15-36-07" src="https://github.com/Automattic/jetpack/assets/60262784/63a78767-68ff-4753-b90a-cfe7c9a0b237">

* Activate the module and check that the form works as expected.
